oeps/oep-0005.rst

```diff
@@ -0,0 +1,187 @@
+=======================================
+OEP-5: Pre-built Developer Environments
+=======================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-5 </oeps/oep-0005>`             |
++---------------+-------------------------------------------+
+| Title         | Pre-built Developer Environments          |
++---------------+-------------------------------------------+
+| Last Modified | 2016-07-20                                |
++---------------+-------------------------------------------+
+| Author        | Calen Pennington <cale@edx.org>           |
++---------------+-------------------------------------------+
+| Arbiter       | Jesse Zoldak <jzoldak@edx.org>            |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+| Created       | 2016-06-28                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+Open edX is a large ecosystem of smaller services. In order to facilitate
+newcomers working in that system, each service will provide a prebuilt environment
+that is packaged, published, and ready to be used for development.
+
+Motivation
+==========
+
+Open edX and the edx-platform repository have gone through several iterations
+of developer setup instructions.
+
+The first set of instructions detailed how to install edx-platform and all of
+its dependencies on your local workstation. As the number of developers and
+developer environments increased, so did the workload of maintaining instructions
+to work around all of the idiosyncrasies in those environments. These instructions
+weren't useful for production installations, which were standardized on a
+particular release of the Ubuntu operating system.
+
+The second, current, method of development is referred to as Devstack. This is
```

Comment:
@cpennington  While I'm generally +1 on this proposal, I'm -2 on the naming convention.

Here, you remind us that "devstack" means something very specific in the OpenEdX context;
it refers to the current, Vagrant-based virtual machine.. 

To redefine this term seems needlessly ripe for confusion.
If we're going to advocate a new development pattern, it needs to have a different name.

I can already see new developers trying to troubleshoot this:
1. $devstack breaks
2. $user googles "fix openedx devstack"
3. $user wades through _years_ of edx-code posts, read-the-docs, and Confluence wiki entries that describe an unrelated, deprecated process.
4. $user gets lucky and realizes "oh not _that_ devstack, the _other_ devstack"
5. ![wtf](https://pbs.twimg.com/profile_images/486977343953440768/LD2fbIF7_400x400.jpeg)